src/bio/bio_monitor.c

```diff
@@ -67,6 +67,25 @@ bio_get_dev_state_internal(void *msg_arg)
 	ABT_eventual_set(dsm->eventual, NULL, 0);
 }
 
+static void
+bio_set_faulty_state_internal(void *msg_arg)
+{
+	struct dev_state_msg_arg	*dsm = msg_arg;
+	int				 rc;
+
+	D_ASSERT(dsm != NULL);
+
+	rc = bio_bs_state_set(dsm->xs->bxc_blobstore, BIO_BS_STATE_FAULTY);
+	if (rc)
+		D_ERROR("BIO FAULTY state set failed, rc=%d\n", rc);
+
+	rc = bio_bs_state_transit(dsm->xs->bxc_blobstore);
+	if (rc)
+		D_ERROR("State transition failed, rc=%d\n", rc);
+
+	ABT_eventual_set(dsm->eventual, NULL, 0);
```

Comment:
Should we return error code by eventual?

Reply:
yes that will probably be useful for this case